### ISSUE TITLE
Show diagnostic for `}` token followed by `else` in let else statement

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#14221]
`}` braces before `else` in `let ... else` statements are not permitted, we should show an error here.
```rs
let Some(node) = match expr_or_pat {
    hir_def::expr::ExprOrPatId::ExprId(expr) => expr_node(&body_source_map, expr, &db),
    hir_def::expr::ExprOrPatId::PatId(pat) => pat_node(&body_source_map, pat, &db),
} else { continue; };
```
